dao/src/test/java/se/fortnox/reactivewizard/db/DaoTransactionsTest.java

```diff
@@ -367,6 +370,35 @@ public void shoulUseSpecifiedConnectionAndScheduler() throws SQLException {
         db.verifyConnectionsUsed(1);
     }
 
+    @Test
+    public void shoulUseSecondConnectionAndSchedulerIfFirstObservableHasNoConnectionProvider() throws SQLException {
+        MockDb secondDb = new MockDb();
+        ConnectionProvider otherConnectionProvider = secondDb.getConnectionProvider();
+
+        TestScheduler secondScheduler = new TestScheduler();
+        DbProxy dbProxyWithoutConnectionProvider = new DbProxy(
+            new DatabaseConfig(), secondScheduler,
+            null, new DbStatementFactoryFactory(), new JsonSerializerFactory());
+        DbProxy dbProxyWithConnectionProvider = dbProxyWithoutConnectionProvider.usingConnectionProvider(otherConnectionProvider);
+
+        when(secondDb.getPreparedStatement().executeBatch())
+            .thenReturn(new int[]{1, 1});
+
+        TestDao daoWithConnectionProvider = dbProxyWithConnectionProvider.create(TestDao.class);
+        TestDao daoWithoutConnectionProvider = dbProxyWithoutConnectionProvider.create(TestDao.class);
+
+        daoTransactions.executeTransaction(daoWithoutConnectionProvider.updateSuccess(), daoWithConnectionProvider.updateSuccess())
+            .subscribeOn(secondScheduler).subscribe();
+        secondScheduler.triggerActions();
+
+        secondDb.verifyConnectionsUsed(1);
+        db.verifyConnectionsUsed(0);
+
+        assertThatExceptionOfType(RuntimeException.class)
+            .isThrownBy(() -> daoTransactions.executeTransaction(daoWithoutConnectionProvider.updateSuccess()).subscribe())
```

Comment:
This subscribe should be asynchronous (not blocking) so an exception should be sent to subscriber rather than thrown to caller?